client/state/selectors/get-post-revision-changes.js

```diff
@@ -27,10 +27,16 @@ const getPostRevisionChanges = createSelector(
 		}
 		const previousRevision = orderedRevisions[ revisionIndex + 1 ];
 		const currentRevision = orderedRevisions[ revisionIndex ];
-		return {
+		const diffStartTime = Date.now();
+		const changes = {
 			content: diffKey( 'content', previousRevision, currentRevision ),
 			title: diffKey( 'title', previousRevision, currentRevision ),
 		};
+		const diffEndTime = Date.now();
+
```

Comment:
This seems to be the best place to measure the time taken to generate the diffs but it seems weird to add tracking to a selector especially selectors should aim to be pure...
Would it be reasonable to send the `diffGenerationTime` then track it in the component some how?

Reply:
Interesting that this is in the selector...we're also calling `diffWords` [at the data layer](https://github.com/Automattic/wp-calypso/blob/44332cc3e866a869df0cd1fbaa4b33314dec6057/client/state/data-layer/wpcom/posts/revisions/index.js#L68-L112). Are we diff'ing each post twice?!

Reply:
Perhaps it'd be better to do this diffing in a lib or something similar? We're doing much more than just selecting state here. Perhaps we could even hold the diff data as a slice of state... Either way it feels like this needs some revision :/
I think we may well be diffing multiple times - I'll explore this some more.